### PR TITLE
Add rust support

### PR DESCRIPTION
**Pull request by nellshamrell**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/license_scout/dependency_manager/cargo.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      # - RUBY_ENGINE
+      # - RUBY_ROOT
+      # - RUBY_VERSION
+      #
+      # The original environment restored at the end of this call.
+      #
+      # @param [Proc] block
+      #   the block to execute with the cleaned environment
+      #
+      def with_clean_env(&block)
```
Comment (tduffield) [inline lib/license_scout/dependency_manager/cargo.rb]: Do we still need this?

Comment (nellshamrell) [reply lib/license_scout/dependency_manager/cargo.rb]: Whoops! Removing that!

**Comment by nellshamrell** [comment]
You are right, cargo should be installed as part of the test suite, adding that in.

File: spec/license_scout/dependency_manager/cargo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  end
+
+  describe "#name" do
+    it "equals 'rust_cargo'" do
+      expect(subject.name).to eql("rust_cargo")
+    end
+  end
+
+  describe "#type" do
+    it "equals 'ruby'" do
```
Comment (tduffield) [inline spec/license_scout/dependency_manager/cargo_spec.rb]: should be `equals 'rust'`

Comment (nellshamrell) [reply spec/license_scout/dependency_manager/cargo_spec.rb]: GAH! Good catch!

File: .expeditor/verify.pipeline.yml [hunk 3]
```diff
@@ -1,10 +1,8 @@
 steps:
   - label: ":linux: :ruby:"
     commands:
-      - ruby --version
-      - bundler --version
-      - bundle install
-      - bundle exec rake
+      - chmod +x /workdir/.expeditor/buildkite/verify.sh
```
Comment (scotthain) [inline .expeditor/verify.pipeline.yml]: You can chmod +x on your local machine then check in the verify.sh - the permissions will be updated in github.

File: .expeditor/verify.pipeline.yml [hunk 4]
```diff
@@ -1,10 +1,8 @@
...
+      - /workdir/.expeditor/buildkite/verify.sh
```
Comment (scotthain) [inline .expeditor/verify.pipeline.yml]: usually we have this as `.expeditor/buildkite/verify.sh` - not a required change but that's how most projects do it

File: .expeditor/verify.pipeline.yml (hunk 3 above)
Comment (nellshamrell) [reply .expeditor/verify.pipeline.yml]: For some reason it did not work when I attempted to do it - but I will try it once more

Comment (scotthain) [reply .expeditor/verify.pipeline.yml]: Huh it really should work.

Comment (nellshamrell) [reply .expeditor/verify.pipeline.yml]: trying again now :)

**Pull request merged**